consensus/ibft/src/main/java/tech/pegasys/pantheon/consensus/ibft/validation/MessageValidator.java

```diff
@@ -12,205 +12,27 @@
-      return false;
-    }
-
-    final Block proposedBlock = msg.getPayload().getBlock();
-
-    final Optional<BlockProcessingOutputs> validationResult =
-        blockValidator.validateAndProcessBlock(
-            protocolContext, proposedBlock, HeaderValidationMode.LIGHT, HeaderValidationMode.FULL);
-
-    if (!validationResult.isPresent()) {
-      LOG.info("Invalid Proposal message, block did not pass validation.");
-      return false;
-    }
-
-    return true;
-  }
-
-  private boolean handleSubsequentProposal(
-      final SignedData<ProposalPayload> existingMsg, final SignedData<ProposalPayload> newMsg) {
-    if (!existingMsg.getAuthor().equals(newMsg.getAuthor())) {
-      LOG.debug("Received subsequent invalid Proposal message; sender differs from original.");
-      return false;
-    }
-
-    final ProposalPayload existingData = existingMsg.getPayload();
-    final ProposalPayload newData = newMsg.getPayload();
-
-    if (!proposalMessagesAreIdentical(existingData, newData)) {
-      LOG.debug("Received subsequent invalid Proposal message; content differs from original.");
-      return false;
-    }
-
-    return true;
-  }
-
-  public boolean validatePrepareMessage(final SignedData<PreparePayload> msg) {
-    final String msgType = "Prepare";
-
-    if (!isMessageForCurrentRoundFromValidatorAndProposalAvailable(msg, msgType)) {
-      return false;
-    }
-
-    if (msg.getAuthor().equals(expectedProposer)) {
-      LOG.info("Illegal Prepare message; was sent by the round's proposer.");
-      return false;
-    }
-
-    return validateDigestMatchesProposal(msg.getPayload().getDigest(), msgType);
-  }
-
-  public boolean validateCommmitMessage(final SignedData<CommitPayload> msg) {
-    final String msgType = "Commit";
-
-    if (!isMessageForCurrentRoundFromValidatorAndProposalAvailable(msg, msgType)) {
-      return false;
-    }
-
-    final Block proposedBlock = proposal.get().getPayload().getBlock();
-    final Address commitSealCreator =
-        Util.signatureToAddress(msg.getPayload().getCommitSeal(), proposedBlock.getHash());
-
-    if (!commitSealCreator.equals(msg.getAuthor())) {
-      LOG.info("Invalid Commit message. Seal was not created by the message transmitter.");
-      return false;
-    }
-
-    return validateDigestMatchesProposal(msg.getPayload().getDigest(), msgType);
-  }
-
-  private boolean isMessageForCurrentRoundFromValidatorAndProposalAvailable(
-      final SignedData<? extends Payload> msg, final String msgType) {
-
-    if (!msg.getPayload().getRoundIdentifier().equals(roundIdentifier)) {
-      LOG.info("Invalid {} message, does not match current round.", msgType);
-      return false;
-    }
-
-    if (!validators.contains(msg.getAuthor())) {
-      LOG.info(
-          "Invalid {} message, was not transmitted by a validator for the " + "associated round.",
-          msgType);
-      return false;
-    }
+  private final SignedDataValidator signedDataValidator;
```

Comment:
Perhaps some tests for messageValidator to prove that it is delegating the validation of the payload to the signedDataValidator